SonarQube.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -79,8 +82,27 @@ public override bool Execute()
             pi.FullPath = this.FullProjectPath;
             pi.ProjectLanguage = this.ProjectLanguage;
 
+            string guid = null;
+            if (!String.IsNullOrEmpty(this.ProjectGuid))
+            {
+                guid = this.ProjectGuid;
+            }
+            else if (!String.IsNullOrEmpty(this.SolutionConfigurationContents))
+            {
+                // Try to get GUID from the Solution
+                XmlDocument xmlDoc = new XmlDocument();
```

Comment:
You may be able to simplify this a bit further but this full block could be replaced by:
```
guid = XDocument.Parse(this.SolutionConfigurationContents)
    .Descendants($"ProjectConfiguration")
    .Where(element => element.Attributes("AbsolutePath").Any(attr => attr.Value == this.FullProjectPath))
    .Attributes("Project")
    .Select(attr => attr.Value)
    .FirstOrDefault();
```